mobile/src/main/java/org/openhab/habdroid/core/OpenHabApplication.kt

```diff
@@ -59,13 +79,44 @@ class OpenHabApplication : MultiDexApplication() {
         ConnectionFactory.initialize(this)
         BackgroundTasksManager.initialize(this)
         RemoteLog.initialize(this)
+
+        dataSaverChangeListener?.let { listener ->
+            registerReceiver(listener, IntentFilter(ConnectivityManager.ACTION_RESTRICT_BACKGROUND_CHANGED))
+            isDataSaverActive = listener.isDataSaverEnabled(this)
+        }
     }
 
     override fun onTerminate() {
         super.onTerminate()
         ConnectionFactory.shutdown()
     }
 
+    fun registerDataSaverStateChangedListener(l: OnDataSaverActiveStateChangedListener) {
+        dataSaverListeners.add(l)
+    }
+
+    fun unregisterDataSaverStateChangedListener(l: OnDataSaverActiveStateChangedListener) {
+        dataSaverListeners.remove(l)
+    }
+
+    inner class DataSaverStateChangeReceiver : BroadcastReceiver() {
+        fun isDataSaverEnabled(context: Context): Boolean {
```

Comment:
```suggestion
        fun isSystemDataSaverEnabled(context: Context): Boolean {
```
IMO this should be renamed to be clear that it's the system data saver, not the app pref.

Reply:
Done.